### PR TITLE
core: Add customer type to sender/receiver validation methods and delete calls.

### PR DESCRIPTION
**Pull request by erika-sdf**
<details>
  <summary>PR Checklist</summary>

### PR Structure

* [x] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [x] This PR's title starts with name of package that is most changed in the PR, ex.
  `paymentservice.stellar`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [x] This PR adds tests for the most critical parts of the new functionality or fixes.
</details>

### What

Add customer type to sender/receiver validation methods and delete calls.

### Why

Customer type should be specified for these calls, as specified in SEP12.

### Known limitations

This is meant to be a short term-solution to resolve integration issues. TODOs have been added to address long term concerns (eg, choosing the first customer type in the sender/receiver map)

### CODE REVIEW

File: core/src/main/java/org/stellar/anchor/sep12/Sep12Service.java [hunk 1]
```diff
@@ -68,20 +68,24 @@ public void deleteCustomer(JwtToken jwtToken, String account, String memo, Strin
...
-                .account(account)
-                .memo(memo)
-                .memoType(memoType)
-                .build());
-    if (existingCustomer.getId() == null) {
-      infoF(
-          "No existing customer found for account={} memo={} memoType={}", account, memo, memoType);
-      throw new SepNotFoundException("User not found.");
-    }
+    String senderTypes[] = {"sep31-sender", "sep31-receiver"};
```
Comment (marcelosalloum) [inline core/src/main/java/org/stellar/anchor/sep12/Sep12Service.java:71]:
```suggestion
    String customerTypes[] = {"sep31-sender", "sep31-receiver"};
```

Comment (marcelosalloum) [reply core/src/main/java/org/stellar/anchor/sep12/Sep12Service.java:71]: Maybe for another PR, move it to a config file.

File: core/src/main/java/org/stellar/anchor/sep12/Sep12Service.java [hunk 2]
```diff
@@ -68,20 +68,24 @@ public void deleteCustomer(JwtToken jwtToken, String account, String memo, Strin
...
+    for (String senderType: senderTypes) {
+      Sep12GetCustomerResponse existingCustomer =
+          customerIntegration.getCustomer(
+              Sep12GetCustomerRequest.builder()
+                  .account(account)
+                  .memo(memo)
+                  .memoType(memoType)
+                  .type(senderType)
+                  .build());
+      if (existingCustomer.getId() == null) {
```
Comment (marcelosalloum) [inline core/src/main/java/org/stellar/anchor/sep12/Sep12Service.java:81]: !!  You should only throw a 404 not found if none of the requests in the for loop succeed.

File: core/src/main/java/org/stellar/anchor/sep12/Sep12Service.java (hunk 1 above)
Comment (erika-sdf) [reply core/src/main/java/org/stellar/anchor/sep12/Sep12Service.java]: Acknowledged, added a TODO for this

File: core/src/main/java/org/stellar/anchor/sep12/Sep12Service.java (hunk 2 above)
Comment (erika-sdf) [reply core/src/main/java/org/stellar/anchor/sep12/Sep12Service.java]: Good catch, I'll alter the logic to only error if no match is found.

**Review by marcelosalloum — APPROVED** [review]
LGTM!

**Pull request merged**